staging/src/k8s.io/client-go/transport/round_trippers.go

```diff
@@ -421,8 +421,34 @@ func (rt *debuggingRoundTripper) RoundTrip(req *http.Request) (*http.Response, e
 	}
 	if rt.levels[debugRequestHeaders] {
 		klog.Infof("Request Headers:")
+		knownAuthTypes := map[string]bool{
+			"bearer":    true,
+			"basic":     true,
+			"negotiate": true,
+		}
 		for key, values := range reqInfo.RequestHeaders {
 			for _, value := range values {
+				// Mask credential content from authorization headers
```

Comment:
this should be split into a function (`maskValue`) called as `value = maskValue(key, value)`, and unit tested